File: Python/Introductory/Permutations.py
#!/usr/bin/env python
# coding: utf-8

# In[14]:


array_size = int(input())
def permutations(array_size):
    if array_size == 3 or array_size == 2:
        print('NO SOLUTION')
    else:
        for i in range(2,array_size + 1,2):
            print(i, end= " ")
        for i in range(1,array_size + 1,2):
            print(i, end=" ")
            
permutations(array_size)

File: C++/Introductory/Permutations.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin >> n;
    if (n == 3 || n==2)
        cout << "NO SOLUTION" << endl;
    else
    {
        for(int i = 2; i<n+1;i+=2)
            cout << i << " ";
        for(int i = 1; i<n+1;i+=2)
            cout << i << " ";
    }
}